apps/glusterfs/volume_entry.go

```diff
@@ -615,6 +615,12 @@ func eligibleClusters(db *bolt.DB, v *VolumeEntry,
 			switch {
 			case v.Info.Block && c.Info.Block:
 			case !v.Info.Block && c.Info.File:
+			case !(c.Info.Block || c.Info.File):
+				// bad cluster config!
+				err = fmt.Errorf("Cluster %v lacks both block and file flags",
+					clusterId)
+				logger.LogError(err.Error())
+				return err
```

Comment:
I don't think we want this: The administrator might deliberately have mark this cluster to not create any volumes (maybe temporarily). So I suggest to drop this commit.